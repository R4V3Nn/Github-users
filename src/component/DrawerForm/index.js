import React from 'react';
import Form from '@talend/react-forms';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { formSubmit } from '../../actions/userActions';

const formConfig = {
	jsonSchema: {
		type: 'object',
		properties: {
			name: {
				type: 'string',
				title: 'Name',
			},
			html_url: {
				type: 'string',
				title: 'Url',
			},
			description: {
				type: 'string',
				title: 'Description',
			},
			homepage: {
				type: 'string',
				title: 'Homepage',
			},
			size: {
				type: 'number',
				title: 'Size',
			},
			language: {
				type: 'string',
				title: 'language',
			},
		},
	},
};

const DrawerForm = props => {
	const formProps = {
		...formConfig,
		properties: props.item,
	};
	return (<Form data={formProps} onSubmit={(e, data) => props.onFormSubmit(data.formData)} />);
};


const mapStateToProps = state => ({
	userRepos: state.users.userRepos,
	id: state.users.selectedId,
});

const mapDispatchToProps = dispatch => ({
	onFormSubmit: (id, formData) => dispatch(formSubmit(id, formData)),
});

DrawerForm.propTypes = {
	item: PropTypes.object,
	onFormSubmit: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(DrawerForm);
