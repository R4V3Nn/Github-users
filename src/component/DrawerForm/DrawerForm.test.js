import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import store from '../../store';


import DrawerForm from './';

describe('DrawerForm component', () => {
	const props = {
		jsonSchema: {
			type: 'object',
			properties: {
				name: {
					type: 'string',
					title: 'Name',
				},
				html_url: {
					type: 'string',
					title: 'Url',
				},
				description: {
					type: 'string',
					title: 'Description',
				},
				homepage: {
					type: 'string',
					title: 'Homepage',
				},
				size: {
					type: 'number',
					title: 'Size',
				},
				language: {
					type: 'string',
					title: 'language',
				},
			},
		},
		onSubmit: jest.fn(),

	};


	it('should render DrawerForm component', () => {
		// when
		const wrapper = mount(
			<Provider store={store}>
				<DrawerForm {...props} />
			</Provider>
			);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
