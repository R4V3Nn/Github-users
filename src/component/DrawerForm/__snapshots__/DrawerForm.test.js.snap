// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerForm component should render DrawerForm component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <Connect(DrawerForm)
    jsonSchema={
      Object {
        "properties": Object {
          "description": Object {
            "title": "Description",
            "type": "string",
          },
          "homepage": Object {
            "title": "Homepage",
            "type": "string",
          },
          "html_url": Object {
            "title": "Url",
            "type": "string",
          },
          "language": Object {
            "title": "language",
            "type": "string",
          },
          "name": Object {
            "title": "Name",
            "type": "string",
          },
          "size": Object {
            "title": "Size",
            "type": "number",
          },
        },
        "type": "object",
      }
    }
    onSubmit={[MockFunction]}
  />
</Provider>
`;
